admin/class-yoast-notification-center.php

```diff
@@ -130,50 +221,333 @@ public function set_transient() {
+	/**
+	 * Return the notifications sorted on type and priority
+	 *
+	 * @return array|Yoast_Notification[] Sorted Notifications
+	 */
+	public function get_sorted_notifications() {
+		$notifications = $this->notifications;
+		if ( empty( $notifications ) ) {
+			return array();
+		}
 
-			$id = $notification->get_id();
+		// Sort by severity, error first.
+		usort( $notifications, array( $this, 'sort_notifications' ) );
 
-			if ( in_array( $id, $ids ) ) {
-				unset( $this->notifications[ $key ] );
-				continue;
+		return $notifications;
+	}
+
+	/**
+	 * Sort on type then priority
+	 *
+	 * @param Yoast_Notification $a Compare with B.
+	 * @param Yoast_Notification $b Compare with A.
+	 *
+	 * @return int 1, 0 or -1 for sorting offset.
+	 */
+	private function sort_notifications( Yoast_Notification $a, Yoast_Notification $b ) {
+		$a_type = $a->get_type();
+		$b_type = $b->get_type();
+
+		if ( $a_type === $b_type ) {
+			$a_priority = $a->get_priority();
+			$b_priority = $b->get_priority();
+
+			if ( $a_priority === $b_priority ) {
+				return 0;
 			}
 
-			$ids[] = $id;
+			return ( ( $a_priority < $b_priority ) ? 1 : -1 );
 		}
 
-		$this->notifications = array_unique( $this->notifications );
+		if ( 'error' === $a_type ) {
+			return -1;
+		}
 
-		// Display notifications.
-		if ( count( $this->notifications ) > 0 ) {
-			foreach ( $this->notifications as $notification ) {
-				echo $notification;
-			}
+		if ( 'error' === $b_type ) {
+			return 1;
 		}
 
-		// Clear the local stored notifications.
-		$this->clear_notifications();
+		return 0;
 	}
 
 	/**
 	 * AJAX display notifications
 	 */
 	public function ajax_get_notifications() {
-
 		// Display the notices.
 		$this->display_notifications();
 
 		// AJAX die.
 		exit;
 	}
+
+	/**
+	 * Remove transient when the plugin is deactivated
+	 */
+	public function deactivate_hook() {
+		$this->clear_notification_conditions();
+		$this->clear_notifications();
+	}
+
+	/**
+	 * Write the notifications to a cookie (hooked on shutdown)
+	 *
+	 * Function renamed to 'update_storage'.
+	 *
+	 * @depreacted 3.2 remove in 3.5
+	 */
+	public function set_transient() {
+	}
+
+	/**
+	 * Save persistent notifications to storage
+	 *
+	 * We need to be able to retrieve these so they can be dismissed at any time during the execution.
+	 *
+	 * @since 3.2
+	 *
+	 * @return void
+	 */
+	public function update_storage() {
+		// Create array with all notifications.
+		$notifications = array();
+
+		// Add each notification as array to $arr_notifications.
+		foreach ( $this->notifications as $notification ) {
+			if ( $notification->is_persistent() ) {
+				$notifications[] = $notification->to_array();
+			}
+		}
+
+		// No notifications to store, clear storage.
+		if ( count( $notifications ) === 0 ) {
+			$this->clear_storage();
+
+			return;
+		}
+
+		// Save the notifications to the storage.
+		update_option( self::STORAGE_KEY, WPSEO_Utils::json_encode( $notifications ), false );
```

Comment:
If we remove the option if there are no notification we can safely autoload this option.